iml-api/src/graphql/stratagem.rs

```diff
@@ -0,0 +1,680 @@
+            &format!("{}-cloudsync-cloudsync", uuid),
+            "created",
+            false,
+            false,
+            &vec!["stratagem.cloudsync".into()],
+            serde_json::json!({
+                "remote": remote,
+                "expression": expression,
+                "action": action,
+            }),
+            fs_id,
+            &context.pg_pool,
+        )
+        .await?;
+
+        let mut jobs: Vec<SendJob<HashMap<String, serde_json::Value>>> = vec![SendJob {
+            class_name: "CreateTaskJob".into(),
+            args: vec![("task_id".into(), serde_json::json!(task.id))]
+                .into_iter()
+                .collect(),
+        }];
+
+        let job_range: Vec<_> = (0..jobs.len()).collect();
+
+        let xs = get_target_hosts_by_fsname(&fsname, &context.pg_pool).await?;
+
+        for x in xs {
+            let path = match x.dev_path {
+                Some(x) => x,
+                None => continue,
+            };
+
+            let cfg = stratagem::StratagemConfig {
+                flist_type: "none".into(),
+                summarize_size: true,
+                device: stratagem::StratagemDevice {
+                    path,
+                    groups: vec!["cloudsync".into()],
+                },
+                groups: vec![stratagem::StratagemGroup {
+                    name: "filesync".into(),
+                    rules: vec![stratagem::StratagemRule {
+                        action: "LAT_SHELL_CMD_FID".into(),
+                        expression: expression.clone(),
+                        argument: "filesync".into(),
+                        counter_name: Some("filesync".into()),
+                    }],
+                }],
+            };
+
+            jobs.push(SendJob {
+                class_name: "ScanMdtJob".into(),
+                args: vec![
+                    ("fqdn".into(), serde_json::to_value(&x.fqdn)?),
+                    ("uuid".into(), serde_json::to_value(&uuid)?),
+                    ("fsname".into(), serde_json::to_value(&fsname)?),
+                    ("config".into(), serde_json::to_value(cfg)?),
+                    (
+                        "depends_on_job_range".into(),
+                        serde_json::to_value(&job_range)?,
+                    ),
+                ]
+                .into_iter()
+                .collect(),
+            })
+        }
+
+        let kwargs: HashMap<String, String> =
+            vec![("message".into(), "Stratagem: Cloudsync".into())]
+                .into_iter()
+                .collect();
+
+        let command_id: i32 = iml_job_scheduler_rpc::call(
+            &context.rabbit_pool.get().await?,
+            "run_jobs",
+            vec![jobs],
+            Some(kwargs),
+        )
+        .map_err(ImlApiError::ImlJobSchedulerRpcError)
+        .await?;
+
+        let command = get_command(&context.pg_pool, command_id).await?;
+
+        Ok(command)
+    }
+    async fn run_fast_file_scan(
```

Comment:
Rename: run_report_and_purge

Reply:
`fast_file_scan` is the blessed name for this feature (I agree report_and_purge has more meaning).